File: project-03-realtime-dashboard/code-java/src/test/java/com/example/demo/CreateDbTest.java
package com.example.demo;

import java.nio.file.Files;
import java.nio.file.Paths;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.Statement;

import org.junit.jupiter.api.Test;

public class CreateDbTest {

    @Test
    public void createDatabaseAndTables() {
        String urlDB = "jdbc:postgresql://localhost:5432/postgres";
        try (Connection conn = DriverManager.getConnection(urlDB, "postgres", "postgres");
             Statement stmt = conn.createStatement()) {
            stmt.executeUpdate("CREATE DATABASE jooq_dashboard");
            System.out.println("jooq_dashboard Database created successfully!");
        } catch (Exception e) {
            System.out.println("DB likely exists: " + e.getMessage());
        }

        String urlTarget = "jdbc:postgresql://localhost:5432/jooq_dashboard";
        try (Connection connTarget = DriverManager.getConnection(urlTarget, "postgres", "postgres");
             Statement stmtTarget = connTarget.createStatement()) {
            String sql = new String(Files.readAllBytes(Paths.get("src/main/resources/db/migration/V1__init_dashboard.sql")));
            stmtTarget.executeUpdate(sql);
            System.out.println("Tables created successfully!");
        } catch (Exception e) {
            throw new RuntimeException("Table creation failed", e);
        }
    }
}

File: project-03-realtime-dashboard/code-kotlin/src/test/kotlin/com/example/demo/CreateDbTest.kt
package com.example.demo

import org.junit.jupiter.api.Test
import java.nio.file.Files
import java.nio.file.Paths
import java.sql.DriverManager

class CreateDbTest {

    @Test
    fun createDatabaseAndTables() {
        val urlDB = "jdbc:postgresql://localhost:5432/postgres"
        try {
            DriverManager.getConnection(urlDB, "postgres", "postgres").use { conn ->
                conn.createStatement().use { stmt ->
                    stmt.executeUpdate("CREATE DATABASE jooq_dashboard")
                    println("jooq_dashboard Database created successfully!")
                }
            }
        } catch (e: Exception) {
            println("DB likely exists: \${e.message}")
        }

        val urlTarget = "jdbc:postgresql://localhost:5432/jooq_dashboard"
        try {
            DriverManager.getConnection(urlTarget, "postgres", "postgres").use { connTarget ->
                connTarget.createStatement().use { stmtTarget ->
                    val sql = String(Files.readAllBytes(Paths.get("src/main/resources/db/migration/V1__init_dashboard.sql")))
                    stmtTarget.executeUpdate(sql)
                    println("Tables created successfully!")
                }
            }
        } catch (e: Exception) {
            throw RuntimeException("Table creation failed", e)
        }
    }
}
